src/MementoContainer/Analysis/PropertyAnalyzer.cs

```diff
@@ -8,13 +8,37 @@
 
 namespace MementoContainer.Analysis
 {
-    internal class PropertyAnalyzer : IPropertyAnalyzer
+    /// <summary>
+    /// Default implementation of IPropertyAnalyzer to analyzes certain inputs (e.g., expressions and objects) and look for properties to be registered in the memento.
+    /// </summary>
+    public class PropertyAnalyzer : IPropertyAnalyzer
     {
+        delegate bool PropertyFilter(MementoClassAttribute mementoClassAttr, PropertyInfo pi, IList<Attribute> propertyAttrs);
+        private readonly PropertyFilter DefaultFilter = (mementoClassAttr, pi, propertyAttrs) =>
+        mementoClassAttr != null
+            ? propertyAttrs.Any(attr => attr is MementoPropertyAttribute) && pi.HasGetAndSet()
+            : propertyAttrs.Any(attr => attr is MementoPropertyAttribute);
+
+        PropertyFilter Filter { get; }
```

Comment:
This property needs to be public, and be initialized through the constructor, e.g.:

```csharp
        public PropertyFilter Filter { get; }

        public PropertyAnalyzer(PropertyFilter filter)
        {
            Filter = filter;
        }

        public PropertyAnalyzer() : this(DefaultFilter)
        {
        }
```

Also, `GetProperties` should be changed to use this `Filter` property. The `PropertySelectorForMementoClass` and `PropertySelectorForNonMementoClass` virtual methods are no longer needed.